server/app.es6

```diff
@@ -1,26 +1,391 @@
+      applicantInfoSecondaryMailingCity: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingCity : null,
+      applicantInfoSecondaryMailingState: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingState : null,
+      applicantInfoSecondaryMailingZIP: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingZIP : null,
+      applicantInfoOrganizationName: req.body.applicantInfo.organizationName,
+      applicantInfoWebsite: req.body.applicantInfo.website,
+      applicantInfoOrgType: req.body.applicantInfo.orgType,
+      applicantInfoSecondaryFirstName: req.body.applicantInfo.secondaryFirstName,
+      applicantInfoSecondaryLastName: req.body.applicantInfo.secondaryLastName,
+      type: req.body.type,
+      noncommercialFieldsActivityDescription: req.body.noncommercialFields.activityDescription,
+      noncommercialFieldsLocationDescription: req.body.noncommercialFields.locationDescription,
+      noncommercialFieldsStartDateTime: req.body.noncommercialFields.startDateTime,
+      noncommercialFieldsEndDateTime: req.body.noncommercialFields.endDateTime,
+      noncommercialFieldsNumberParticipants: req.body.noncommercialFields.numberParticipants,
+      noncommercialFieldsSpectatorCount: req.body.noncommercialFields.spectators,
+      signature: req.body.signature,
+      reasonForReturn: req.body.reasonForReturn
+    }).then((noncommApp) => {
+      req.body['applicationId'] = noncommApp.applicationId;
+      req.body['appControlNumber'] = noncommApp.appControlNumber;
+      res.status(201).json(req.body);
+    }).error((err) => {
+      res.status(500).json(err);
+    });
+  }
+};
+
+let updateApp = (req, res) => {
+  NoncommercialApplication.findOne({ 'where': {app_control_number: req.params.id}}).then(app => {
+    if(app) {
+      app.status = req.body.status;
+      if (app.status === 'Accepted') {
+        sendAcceptedNoncommercialApplicationToMiddleLayer(app,
+          (response) => {
+            app.controlNumber = response.controlNumber;
+            app.save().then(() => {
+              res.status(200).json(util.translateFromDatabaseToJSON(app));
+            });
+          },
+          (error) => {
+            res.status(400).send(error);
+          }
+        );
+      } else {
+        app.save().then(() => {
+          res.status(200).json(util.translateFromDatabaseToJSON(app));
+        });
+      }
+    } else {
+      res.status(404);
+    }
+  });
+};
+
+let getApp = (req, res) => {
+  NoncommercialApplication.findOne({ 'where': {app_control_number: req.params.id}})
+    .then(app => {
+      if(app) {
+        res.status(200).json(util.translateFromDatabaseToJSON(app));
+      } else {
+        res.status(404).send();
+      }
+    })
+    .catch(error => res.status(400).json(error.message));
```

Comment:
This seems like it would be a `500` as it is probably catching internal db errors.